src/v/cloud_storage/cache_service.cc

```diff
@@ -32,13 +33,49 @@
 
 namespace cloud_storage {
 
-cache::cache(std::filesystem::path cache_dir) noexcept
+cache::cache(
+  std::filesystem::path cache_dir,
+  size_t max_cache_size,
+  ss::lowres_clock::duration check_period) noexcept
   : _cache_dir(std::move(cache_dir))
+  , _max_cache_size(max_cache_size)
+  , _check_period(check_period)
   , _cnt(0) {}
 
+ss::future<uint64_t> cache::get_curr_cache_size() {
+    // todo: implement this
+    return seastar::make_ready_future<uint64_t>(0);
+}
+
+ss::future<> cache::clean_up_cache() {
+    uint64_t curr_cache_size = co_await get_curr_cache_size();
```

Comment:
this needs a gate, the operation will take considerable time so we need to ensure that cleanup is stopped when the cache is being stopped

Reply:
Done.